site/documentation/content/api/command-and-control-kafka/index.md

```diff
@@ -0,0 +1,117 @@
+| *message-id*       | no        | *string*  | An identifier that uniquely identifies the message at the sender side. |
+
+The command message MAY contain arbitrary payload, set as message value, to be sent to the device. The value of the message's *subject* header may provide a hint to the device regarding the format, encoding and semantics of the payload data.
+
+## Send a (Request/Response) Command
+
+*Business Applications* use this operation to send a command to a device for which they expect the device to send back a response.
+For that, the Business Application writes a message to the tenant-specific topic `hono.command.${tenant_id}` where `${tenant_id}` is the ID of the tenant that the client wants to send the command for.
+The Business Application can consume the corresponding command response from the `hono.command_response.${tenant_id}` topic.
+
+**Preconditions**
+
+1. Either the topics `hono.command.${tenant_id}` and `hono.command_response.${tenant_id}` exist, or the broker is configured to automatically create topics on demand.
+1. The *Business Application* is authorized to write to the `hono.command.${tenant_id}` topic and read from the `hono.command_response.${tenant_id}` topic.
+1. The *Business Application* subscribes to the `hono.command_response.${tenant_id}` topic with a Kafka consumer.
+
+**Message Flow**
+
+1. The *Business Application* produces a command message to the *Kafka Cluster* on the topic `hono.command.${tenant_id}`.
+1. Hono consumes the message from the *Kafka Cluster* and forwards it to the device, provided that device is connected and accepting commands.
+1. The device sends a command response message. Hono writes that message into the `hono.command_response.${tenant_id}` topic on the *Kafka Cluster*.
+1. The *Business Application* consumes the command response message from the `hono.command_response.${tenant_id}` topic.
+
+**Command Message Format**
+
+The key of the message MUST be the ID of the device that the command is targeted at.
+
+Metadata MUST be set as Kafka headers on a message.
+The following table provides an overview of the headers the *Business Application* needs to set on a request/response command message.
+
+| Name               | Mandatory | Type      | Description |
+| :----------------- | :-------: | :-------- | :---------- |
+| *device_id*        | yes       | *string*  | The identifier of the device that the command is targeted at. |
+| *subject*          | yes       | *string*  | The name of the command to be executed by the device. |
+| *expect-response*  | yes       | *boolean* | MUST be set with a value of `true` for a command for which a response from the device is expected. |
```

Comment:
why don't we simply use the presence of the *correlation-id* for this purpose?

Reply:
I first I thought it would be better to not just use the `correlation-id` (which could be routinely set for all messages by the northbound application) but instead make it more explicit if actually a response is expected.

But looking more closely here, marking the command as a request/response command doesn't have much of an impact anyway. It's mostly used for the command metrics.
So, yes, I'll remove the `expect-response` header.

Reply:
@sophokles73 I've revisited this topic here and now think, that we should indeed introduce some distinguishing property other than the `correlation-id`.

The `correlation-id` property is something that should possibly correlate a one-way command message with the previous telemetry/event `ttd` message that triggered the command. That would be the straightforward way to implement a device request/response communication pattern (device to application/cloud - see #1276).

Therefore I would want to have a separate boolean property, making it explicit that a command-response is expected.

I would adapt the name slightly and choose `response-expected`.

Other options:
`response-required` - doesn't fit that well because it would imply that a timeout/error is thrown if no response is received (which isn't the case)
`one-way` - omitting the property would mean the request/response scheme as default, that would be different from AMQP, where the default without added property is one-way

Reply:
using `response-expected` sounds good to me

Reply:
Ok, I've added that.
I've also added a commit to hide the menu entry for now. When it is clear that the spec implementation will be in the next Hono release, that commit can be reverted.

Reply:
Do you really want to make it mandatory? We could also make it optional and let it default to `false` (saving a few bytes in the message) ...

Reply:
It's marked as mandatory in the table above because that is in the "Send a (Request/Response) Command" chapter.
In the headers table for the "Send a One-Way Command" chapter, the header isn't listed.

So, one-way commands do not have to have a response-expected=false header.

A combined headers table for both kinds of commands would make this more straightforward, I guess. But with the structure of having separate one-way and request-response chapters (which makes sense to me, and is also used in the AMQP C&C spec), this would look like duplicate information.